tests/test_date.py

```diff
@@ -118,163 +192,230 @@ def test_should_one_date_for_each_month(self, low, high, expected_results):
                                 datetime(2014, 12, 31, 23, 59, 59, 0)]),
     ])
     def test_periods(self, low, high, period, expected_results):
-        results = list(date.get_intersecting_periods(low, high, period=period))
-        self.assertEquals(expected_results, results)
+        self.given_period_low(low)
+        self.given_period_high(high)
+        self.given_period(period)
+        self.given_expected_results(expected_results)
+        self.when_intersecting_period_calculated()
+        self.then_should_one_date_for_each_month()
+
+    def test_should_reject_easily_mistaken_dateutil_arguments(self):
+        periods = [
+            'years',
+            'months',
+            'days',
+            'hours',
+            'minutes',
+            'seconds',
+            'microseconds',
+            'some_period',
+        ]
+        self.given_period_low(datetime(2014, 6, 15))
+        self.given_period(datetime(2014, 6, 25))
```

Comment:
There is true developer who counts objects from zero in real life :smiley: 
Second one is actually `high`. So you should use `self.given_period_high` here.

I understand that original incorrect code may be confusing, but you need to be able to distinguish right from wrong and make a change (or at least suggestion). Everybody makes mistakes. We work in teams to minimize the damage one individual can do. Here is famous [Boy Scout Rule](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule) by Uncle Bob. :)